codegen/generate.go

```diff
@@ -65,21 +81,32 @@ type GenerateConfig struct {
 // - if ExistsSkip, do nothing and return
 // - if ExistsOverwrite, overwrite the existing file
 func WriteToFile(terragruntOptions *options.TerragruntOptions, basePath string, config GenerateConfig) error {
-	// If this GenerateConfig is disabled then skip further processing.
-	if config.Disable {
-		terragruntOptions.Logger.Debugf("Skipping generating file at %s because it is disabled", config.Path)
-		return nil
-	}
-
 	// Figure out thee target path to generate the code in. If relative, merge with basePath.
 	var targetPath string
 	if filepath.IsAbs(config.Path) {
 		targetPath = config.Path
 	} else {
 		targetPath = filepath.Join(basePath, config.Path)
 	}
-
 	targetFileExists := util.FileExists(targetPath)
+
+	// If this GenerateConfig is disabled then skip further processing.
+	if config.Disable {
+		terragruntOptions.Logger.Debugf("Skipping generating file at %s because it is disabled", config.Path)
+
+		if targetFileExists {
+			if shouldRemove, err := shouldRemoveWithFileExists(terragruntOptions, targetPath, config.IfDisabled); err != nil {
+				return err
```

Comment:
Make sense to wrap in `errors.WithStackTrace(err)`?  same for other places where err is returned

Reply:
> Make sense to wrap in `errors.WithStackTrace(err)`? same for other places where err is returned

No, we only have to wrap the call from the 3rd party library, `shouldRemoveWithFileExists` already has our error wrappers, if we wrap here the stack trace will start here and not from inside `mustRemoveWithFileExists`.